### DISCUSSION TITLE
Debug Service job fails to start -  qsh: 001-0019 Error found searching for command undefined. No such path or directory.

### DISCUSSION THREAD

**Discussion by heymchri** [codefori/vscode-ibmi#2216] [IBM i]
I'm setting up the Debug Service on a V7R4 partition.
Installed the prereq PTF SI86178, the Open Source Env, made bash the default shell, set up PATH in /QOpenSys/etc/profile.local, generated certs, but when I try to start the Debug Service job, it fails and produces a QPRINT file with the following error messsage:
qsh: 001-0019 Error found searching for command undefined. No such path or directory.

Starting the Debug Service job from a 5250 session works OK:
SBMJOB CMD(STRQSH CMD('/QOpenSys/pkgs/bin/bash -c /QIBM/ProdData/IBMiDebugService/bin/startDebugService.sh')) JOB(DBGSVCE) JOBQ(QSYS/QUSRNOMAX) JOBD(QSYS/QSYSJOBD) USER(*CURRENT)

Any ideas why I can't start it from VS Code?

IBM i Debug v2.0.1

**Comment by sebjulliand** [comment]
Hi!
Try downloading the logs from here and attach them here:
![image](https://github.com/user-attachments/assets/3c7d836e-e4e7-4614-bc4d-2447b92d0c31)

There may be some hints in the Code for i output.

**Reply by heymchri** [reply]
I have attached the logs.
In the CodeForBMi.txt file, you can see that my default shell is /QOpenSys/pkgs/bin/bash, but towards the end of the file, you can see this: 
/QIBM/UserData/IBMiDebugService: /QOpenSys/usr/bin/qsh
system  "SBMJOB CMD(STRQSH CMD('undefined -c /QIBM/ProdData/IBMiDebugService/bin/startDebugService.sh')) JOB(DBGSVCE) JOBQ(QSYS/QUSRNOMAX) JOBD(QSYS/QSYSJOBD) USER(*CURRENT)"

[CodeForIBMi_20240814_094820.zip](https://github.com/user-attachments/files/16609481/CodeForIBMi_20240814_094820.zip)

**Reply by heymchri** [reply]
Hi, it looks like I was missing .bashrc in my home folder - after adding this with the following contents, I was able to start the Debug Service from within VS Code:
export PATH=/QOpenSys/pkgs/bin:$PATH
While testing different scenarios with and without this file present, at one point I did get a message upon connecting to the system that /QOpenSys/pkg/bin was not found in $PATH, and it proposed to create the .bashrc file in my home directory. 
There seem to be a couple of issues related to this:

1. this message does not always show up when the .bashrc file does not exist in my home directory (I made sure to restart VS Code before every test)
2. even without the .bashrc file present in my home directory, when I connect through SSH and look at my $PATH, it does show /QOpenSys/pkgs/bin as part of it:
~ % ssh heymchr@xxxxxxx
heymchr@xxxxxxxx's password:
-bash-5.2$ echo $PATH
/QOpenSys/pkgs/bin:/QOpenSys/usr/bin:/usr/ccs/bin:/QOpenSys/usr/bin/X11:/usr/sbin:.:/usr/bin
3. Now, even after deleting the .bashrc file, I can still start the Debug Service job from VS Code (I made sure to restart VS Code before testing this)

Long story short, everything seems to be working now, but I'm not quite sure why I ran into the problem in the first place...

**Comment by sebjulliand** [comment]
This `undefined` here:
```shell
SBMJOB CMD(STRQSH CMD('undefined -c /QIBM/ProdData/IBMiDebugService/bin/startDebugService.sh')) JOB(DBGSVCE) JOBQ(QSYS/QUSRNOMAX) JOBD(QSYS/QSYSJOBD) USER(*CURRENT)
```
It seems to indicate that `bash` was not installed by the time you first connected to this LPAR. Since Code for IBM i check and then caches the features installed for each connection (so it doesn't do it each time, to not impede connection time), the `bash` feature was `undefined` for this connection even after it was installed.

You might have connected at some point using this option:
![image](https://github.com/user-attachments/assets/0054152d-22d9-4879-bc8c-47c1e05550ac)

This makes Code for i reload every features and configuration value that are cached. It made `bash` marked as available, hence resolving the issue faced when trying to start the service.

But that's only an educated guess 😅
